File: potd-gfg/Positive Negative Pair (9th January 2022)/PositiveNegativePair.java
Problem Link: 
https://practice.geeksforgeeks.org/problems/positive-negative-pair5209/1

class Solution
{
    public static ArrayList<Integer>findPairs(int arr[], int n) 
        { 
            // code here.
            ArrayList<Integer> ans=new ArrayList<Integer>();
            HashSet<Integer> hs=new HashSet<>();
            for(int i=0;i<n;i++){
                if(hs.contains(Math.abs(arr[i]))){
                    ans.add(-Math.abs(arr[i]));
                    ans.add(Math.abs(arr[i]));
                }
                hs.add(Math.abs(arr[i]));
            }

            return ans;
        }
}

File: potd-gfg/Positive Negative Pair (9th January 2022)/PositiveNegativePair.py
#User function Template for python3
# problem link https://practice.geeksforgeeks.org/problems/positive-negative-pair5209/1#

class Solution:
    #Function to return list containing all the pairs having both
    #negative and positive values of a number in the array.
    def findPairs(self,arr,n):
        # code here 
        st = set()
        res = []
        
        for i in arr:
            positive = abs(i)
            negative = -abs(i)
            if positive in st:
                res.extend([negative, positive])
            else:
                st.add(positive)
        return res
